### PR TITLE
Create hochschule-fur-soziale-arbeit-fhnw.csl

### PR DESCRIPTION
**Pull request by JoshuaWeberFHNW**
Dear Community,

I´m new into github and CSL and i have tried to create a citation style based on the citation-guide from the university of applied science i´m working for. I would appreciate, if the style could be added to the csl-style-repository, so that students and colleagues may use it.

Thanks in advance!

Kind regards,
Joshua Weber

### CODE REVIEW

File: hochschule-fur-soziale-arbeit-fhnw.csl [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+        <choose>
+          <if variable="URL" match="any">
+            <text variable="URL" prefix="URL: "/>
+            <date variable="accessed" prefix=" [Zugriffsdatum: " suffix="].">
+              <date-part form="numeric-leading-zeros" name="day" suffix=". "/>
+              <date-part form="long" name="month" suffix=" "/>
+              <date-part name="year"/>
+            </date>
+          </if>
+          <else-if variable="DOI" match="any">
```
Comment (adam3smith) [inline hochschule-fur-soziale-arbeit-fhnw.csl]: Is this really what the guide wants? Preferring a URL over a DOI seems misguided to me. DOIs are permanent. URLs aren't necessarily.

**Comment by JoshuaWeberFHNW** [comment]
Dear Sebastian

Thank you so much for the helpful hints you gave to me! I will work through your questions the next days and let you know as soon as I am ready.

Best regards
Joshua Weber

Von: Sebastian Karcher <notifications@github.com>
Gesendet: Sonntag, 21. Oktober 2018 03:01
An: citation-style-language/styles <styles@noreply.github.com>
Cc: Weber Joshua <joshua.weber@fhnw.ch>; Author <author@noreply.github.com>
Betreff: Re: [citation-style-language/styles] Create hochschule-fur-soziale-arbeit-fhnw.csl (#3750)


@adam3smith requested changes on this pull request.

Some questions and change requests. Let us know if you have any questions.

________________________________

In hochschule-fur-soziale-arbeit-fhnw.csl<https://github.com/citation-style-language/styles/pull/3750#discussion_r226844474>:

> @@ -0,0 +1,247 @@

+<?xml version="1.0" encoding="utf-8"?>

+<style xmlns="http://purl.org/net/xbiblio/csl" class="in-text" version="1.0" demote-non-dropping-particle="sort-only" default-locale="de-CH">

+  

+  <info>

+    <title>Hochschule für Soziale Arbeit FHNW (German - Switzerland)</title>

+    <title-short>HSA FHNW</title-short>

+    <id>http://www.zotero.org/styles/hochschule-fur-soziale-arbeit-fhnw</id<http://www.zotero.org/styles/hochschule-fur-soziale-arbeit-fhnw%3c/id>>

+    <link href="http://www.zotero.org/styles/hochschule-fur-soziale-arbeit-fhnw" rel="self"/>

+    <link href="https://welcome.inside.fhnw.ch/organisation/hochschule/HSA/Studierende/Freigegebene%20Dokumente/Wegleitung_Gestaltung_wissenschaftliches_Arbeiten.pdf" rel="documentation"/>

That's an intranet link -- anyway to make this available without registration (if need be, by uploading it to this pull request)?

________________________________

In hochschule-fur-soziale-arbeit-fhnw.csl<https://github.com/citation-style-language/styles/pull/3750#discussion_r226844486>:

> +        <names variable="editor"/>

+      </substitute>

+    </names>

+  </macro>

+  

+  <macro name="contributors">

+    <choose>

+      <if variable="author" match="any">

+        <names variable="author">

+          <name delimiter="/" delimiter-precedes-last="never" name-as-sort-order="all"/>

+          <label form="verb-short" prefix=" " text-case="lowercase"/>

+        </names>

+      </if>

+      <else-if variable="editor" match="any">

+        <names variable="editor">

+          <name form="short" delimiter="/" et-al-min="4" et-al-use-first="1" et-al-subsequent-min="3" et-al-subsequent-use-first="1"/>

I'm thinking the et-al for editors in the bibliography is an oversight?

________________________________

In hochschule-fur-soziale-arbeit-fhnw.csl<https://github.com/citation-style-language/styles/pull/3750#discussion_r226844498>:

> +          <text term="in" text-case="capitalize-first"/>

+          <text macro="container-contributors" suffix=". "/>

+        </group>

+        <text variable="container-title"/>

+      </if>

+      <else-if type="book article-journal entry-dictionary entry-encyclopedia report" match="any">

+        <group delimiter=": ">

+          <text term="in" text-case="capitalize-first"/>

+          <text variable="container-title" suffix="."/>

+        </group>

+      </else-if>

+      <else-if type="article-newspaper article-magazine" match="any">

+        <group>

+          <date variable="accessed" prefix=" Erschienen am ">

+            <date-part form="numeric-leading-zeros" name="day" suffix=". "/>

+            <date-part form="long" name="month" suffix=" "/>

Note I have removed the prefix from all months in dates. Prefixes don't work well on date parts, as you don't know whether the previous part (day in this case) is or isn't present.

________________________________

In hochschule-fur-soziale-arbeit-fhnw.csl<https://github.com/citation-style-language/styles/pull/3750#discussion_r226844506>:

> +    <choose>

+      <if type="speech" match="any">

+        <date variable="issued" prefix=" Präsentation am ">

+          <date-part form="numeric-leading-zeros" name="day" suffix=". "/>

+          <date-part form="long" name="month" suffix=" "/>

+          <date-part name="year" suffix=". "/>

+        </date>

+        <text variable="event" suffix="."/>

+        <text variable="event-place" prefix=" " suffix="."/>

+      </if>

+    </choose>

+  </macro>

+  

+  <macro name="container-contributors">

+    <choose>

+      <if type="chapter" match="any">

You can almost certainly simplify this significantly using substitute.

________________________________

In hochschule-fur-soziale-arbeit-fhnw.csl<https://github.com/citation-style-language/styles/pull/3750#discussion_r226844522>:

> +          <choose>

+            <if variable="author editor" match="any">

+              <names variable="translator">

+                <name form="long" name-as-sort-order="all" and="text" delimiter="; " delimiter-precedes-last="never"/>

+                <label form="short" prefix=" (" suffix=")"/>

+              </names>

+            </if>

+          </choose>

+        </group>

+      </if>

+    </choose>

+  </macro>

+  

+  <macro name="edition">

+    <choose>

+      <if variable="edition" match="any">

no need to test for edition first. If it's not present, CSL just won't print the next line. Also, please use the edition term instead of "Aufl. " in a suffix.

________________________________

In hochschule-fur-soziale-arbeit-fhnw.csl<https://github.com/citation-style-language/styles/pull/3750#discussion_r226844530>:

> +    </choose>

+  </macro>

+  

+  <macro name="edition">

+    <choose>

+      <if variable="edition" match="any">

+        <text variable="edition" form="short" prefix=". " suffix=". Aufl. "/>

+      </if>

+    </choose>

+  </macro>

+  

+  <macro name="publisher">

+    <choose>

+      <if type="book chapter report paper-conference" match="any">

+        <group delimiter=": ">

+          <text variable="publisher-place" prefix=" "/>

makes more sense to set the prefix and suffix on the group, so that they'll work even if publisher or place are missing. Same for thesis below.

________________________________

In hochschule-fur-soziale-arbeit-fhnw.csl<https://github.com/citation-style-language/styles/pull/3750#discussion_r226844536>:

> +      </if>

+      <else-if type="thesis">

+        <group delimiter=". ">

+          <text variable="publisher" prefix=" "/>

+          <text variable="publisher-place" suffix="."/>

+        </group>

+      </else-if>

+    </choose>

+  </macro>

+  

+  <macro name="locators">

+    <choose>

+      <if type="article article-journal article-magazine chapter">

+        <text variable="volume" prefix=" " suffix=". Jg."/>

+        <text variable="issue" prefix=" (" suffix=")."/>

+        <text variable="page" prefix=" S. " suffix=""/>

preferable to use term rather than prefix for page.

________________________________

In hochschule-fur-soziale-arbeit-fhnw.csl<https://github.com/citation-style-language/styles/pull/3750#discussion_r226844550>:

> +    </choose>

+  </macro>

+  

+  <macro name="access">

+    <choose>

+      <if type="book chapter" match="none">

+        <choose>

+          <if variable="URL" match="any">

+            <text variable="URL" prefix="URL: "/>

+            <date variable="accessed" prefix=" [Zugriffsdatum: " suffix="].">

+              <date-part form="numeric-leading-zeros" name="day" suffix=". "/>

+              <date-part form="long" name="month" suffix=" "/>

+              <date-part name="year"/>

+            </date>

+          </if>

+          <else-if variable="DOI" match="any">

Is this really what the guide wants? Preferring a URL over a DOI seems misguided to me. DOIs are permanent. URLs aren't necessarily.

**Comment by JoshuaWeberFHNW** [comment]
Dear Sebastian Karcher

I now was able to handle the requested changes. I accounted all your requests and attached the revised CSL-style to this mail. Let me know if there is still something to correct/improve.

Kind regards
Joshua Weber

Von: Sebastian Karcher <notifications@github.com>
Gesendet: Sonntag, 21. Oktober 2018 03:01
An: citation-style-language/styles <styles@noreply.github.com>
Cc: Weber Joshua <joshua.weber@fhnw.ch>; Author <author@noreply.github.com>
Betreff: Re: [citation-style-language/styles] Create hochschule-fur-soziale-arbeit-fhnw.csl (#3750)


@adam3smith requested changes on this pull request.

Some questions and change requests. Let us know if you have any questions.

________________________________

In hochschule-fur-soziale-arbeit-fhnw.csl<https://github.com/citation-style-language/styles/pull/3750#discussion_r226844474>:

> @@ -0,0 +1,247 @@

+<?xml version="1.0" encoding="utf-8"?>

+<style xmlns="http://purl.org/net/xbiblio/csl" class="in-text" version="1.0" demote-non-dropping-particle="sort-only" default-locale="de-CH">

+  

+  <info>

+    <title>Hochschule für Soziale Arbeit FHNW (German - Switzerland)</title>

+    <title-short>HSA FHNW</title-short>

+    <id>http://www.zotero.org/styles/hochschule-fur-soziale-arbeit-fhnw</id<http://www.zotero.org/styles/hochschule-fur-soziale-arbeit-fhnw%3c/id>>

+    <link href="http://www.zotero.org/styles/hochschule-fur-soziale-arbeit-fhnw" rel="self"/>

+    <link href="https://welcome.inside.fhnw.ch/organisation/hochschule/HSA/Studierende/Freigegebene%20Dokumente/Wegleitung_Gestaltung_wissenschaftliches_Arbeiten.pdf" rel="documentation"/>

That's an intranet link -- anyway to make this available without registration (if need be, by uploading it to this pull request)?

________________________________

In hochschule-fur-soziale-arbeit-fhnw.csl<https://github.com/citation-style-language/styles/pull/3750#discussion_r226844486>:

> +        <names variable="editor"/>

+      </substitute>

+    </names>

+  </macro>

+  

+  <macro name="contributors">

+    <choose>

+      <if variable="author" match="any">

+        <names variable="author">

+          <name delimiter="/" delimiter-precedes-last="never" name-as-sort-order="all"/>

+          <label form="verb-short" prefix=" " text-case="lowercase"/>

+        </names>

+      </if>

+      <else-if variable="editor" match="any">

+        <names variable="editor">

+          <name form="short" delimiter="/" et-al-min="4" et-al-use-first="1" et-al-subsequent-min="3" et-al-subsequent-use-first="1"/>

I'm thinking the et-al for editors in the bibliography is an oversight?

________________________________

In hochschule-fur-soziale-arbeit-fhnw.csl<https://github.com/citation-style-language/styles/pull/3750#discussion_r226844498>:

> +          <text term="in" text-case="capitalize-first"/>

+          <text macro="container-contributors" suffix=". "/>

+        </group>

+        <text variable="container-title"/>

+      </if>

+      <else-if type="book article-journal entry-dictionary entry-encyclopedia report" match="any">

+        <group delimiter=": ">

+          <text term="in" text-case="capitalize-first"/>

+          <text variable="container-title" suffix="."/>

+        </group>

+      </else-if>

+      <else-if type="article-newspaper article-magazine" match="any">

+        <group>

+          <date variable="accessed" prefix=" Erschienen am ">

+            <date-part form="numeric-leading-zeros" name="day" suffix=". "/>

+            <date-part form="long" name="month" suffix=" "/>

Note I have removed the prefix from all months in dates. Prefixes don't work well on date parts, as you don't know whether the previous part (day in this case) is or isn't present.

________________________________

In hochschule-fur-soziale-arbeit-fhnw.csl<https://github.com/citation-style-language/styles/pull/3750#discussion_r226844506>:

> +    <choose>

+      <if type="speech" match="any">

+        <date variable="issued" prefix=" Präsentation am ">

+          <date-part form="numeric-leading-zeros" name="day" suffix=". "/>

+          <date-part form="long" name="month" suffix=" "/>

+          <date-part name="year" suffix=". "/>

+        </date>

+        <text variable="event" suffix="."/>

+        <text variable="event-place" prefix=" " suffix="."/>

+      </if>

+    </choose>

+  </macro>

+  

+  <macro name="container-contributors">

+    <choose>

+      <if type="chapter" match="any">

You can almost certainly simplify this significantly using substitute.

________________________________

In hochschule-fur-soziale-arbeit-fhnw.csl<https://github.com/citation-style-language/styles/pull/3750#discussion_r226844522>:

> +          <choose>

+            <if variable="author editor" match="any">

+              <names variable="translator">

+                <name form="long" name-as-sort-order="all" and="text" delimiter="; " delimiter-precedes-last="never"/>

+                <label form="short" prefix=" (" suffix=")"/>

+              </names>

+            </if>

+          </choose>

+        </group>

+      </if>

+    </choose>

+  </macro>

+  

+  <macro name="edition">

+    <choose>

+      <if variable="edition" match="any">

no need to test for edition first. If it's not present, CSL just won't print the next line. Also, please use the edition term instead of "Aufl. " in a suffix.

________________________________

In hochschule-fur-soziale-arbeit-fhnw.csl<https://github.com/citation-style-language/styles/pull/3750#discussion_r226844530>:

> +    </choose>

+  </macro>

+  

+  <macro name="edition">

+    <choose>

+      <if variable="edition" match="any">

+        <text variable="edition" form="short" prefix=". " suffix=". Aufl. "/>

+      </if>

+    </choose>

+  </macro>

+  

+  <macro name="publisher">

+    <choose>

+      <if type="book chapter report paper-conference" match="any">

+        <group delimiter=": ">

+          <text variable="publisher-place" prefix=" "/>

makes more sense to set the prefix and suffix on the group, so that they'll work even if publisher or place are missing. Same for thesis below.

________________________________

In hochschule-fur-soziale-arbeit-fhnw.csl<https://github.com/citation-style-language/styles/pull/3750#discussion_r226844536>:

> +      </if>

+      <else-if type="thesis">

+        <group delimiter=". ">

+          <text variable="publisher" prefix=" "/>

+          <text variable="publisher-place" suffix="."/>

+        </group>

+      </else-if>

+    </choose>

+  </macro>

+  

+  <macro name="locators">

+    <choose>

+      <if type="article article-journal article-magazine chapter">

+        <text variable="volume" prefix=" " suffix=". Jg."/>

+        <text variable="issue" prefix=" (" suffix=")."/>

+        <text variable="page" prefix=" S. " suffix=""/>

preferable to use term rather than prefix for page.

________________________________

In hochschule-fur-soziale-arbeit-fhnw.csl<https://github.com/citation-style-language/styles/pull/3750#discussion_r226844550>:

> +    </choose>

+  </macro>

+  

+  <macro name="access">

+    <choose>

+      <if type="book chapter" match="none">

+        <choose>

+          <if variable="URL" match="any">

+            <text variable="URL" prefix="URL: "/>

+            <date variable="accessed" prefix=" [Zugriffsdatum: " suffix="].">

+              <date-part form="numeric-leading-zeros" name="day" suffix=". "/>

+              <date-part form="long" name="month" suffix=" "/>

+              <date-part name="year"/>

+            </date>

+          </if>

+          <else-if variable="DOI" match="any">

Is this really what the guide wants? Preferring a URL over a DOI seems misguided to me. DOIs are permanent. URLs aren't necessarily.

**Comment by JoshuaWeberFHNW** [comment]
Hi Sebastian Karcher,

Thanks! I pasted the edited style into the register “Files changed”.

Best regards
Joshua Weber

Von: Sebastian Karcher <notifications@github.com>
Gesendet: Donnerstag, 8. November 2018 04:28
An: citation-style-language/styles <styles@noreply.github.com>
Cc: Weber Joshua <joshua.weber@fhnw.ch>; Mention <mention@noreply.github.com>
Betreff: Re: [citation-style-language/styles] Create hochschule-fur-soziale-arbeit-fhnw.csl (#3750)


Hi @JoshuaWeberFHNW<https://github.com/JoshuaWeberFHNW> ,

you can't attach files here. Please return to the ticket on github (#3750<https://github.com/citation-style-language/styles/pull/3750> ) and update you're style there as described in the first automated message<https://github.com/citation-style-language/styles/pull/3750#issuecomment-430987982> on your post.

**Pull request merged**